File: week_1/day-3-two-pointer/two_pointers_warmup/container_with_most_water.py
height = [1,8,6,2,5,4,8,3,7]

def max_area(height):
    left, right = 0, len(height) - 1
    max_area = 0

    while left < right:
        area = (right - left) * min(height[left], height[right])
        max_area = max(max_area, area)

        if height[left] < height[right]:
            left += 1
        else:
            right -= 1

    return max_area

print(max_area(height))  # Output: 49

File: week_1/day-3-two-pointer/two_pointers_warmup/container_with_most_water.js
const height = [1,8,6,2,5,4,8,3,7];

function maxArea(height) {
    let left = 0, right = height.length - 1;
    let maxArea = 0;

    while (left < right) {
        const area = (right - left) * Math.min(height[left], height[right]);
        maxArea = Math.max(maxArea, area);

        height[left] < height[right] ? left++ : right--;
    }
    return maxArea;
}

console.log(maxArea(height)); // 49
